examples/rust/components/blobby/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+#![allow(clippy::missing_safety_doc)]
+// ^^^ This is for clippy complaining about something that wit bindgen is generating
+
+wit_bindgen::generate!({
+    world: "blobby",
+});
+
+use std::io::{Read, Write};
+
+use http::{
+    header::{ALLOW, CONTENT_LENGTH},
+    StatusCode,
+};
+
+use exports::wasi::http::incoming_handler::Guest;
+use wasi::blobstore::blobstore;
+use wasi::http::types::*;
+use wasi::logging::logging::{log, Level};
+use wrapper::OutputStreamWriter;
+
+mod wrapper;
+
+struct Error {
+    status_code: StatusCode,
+    message: String,
+}
+
+impl Error {
+    fn from_blobstore_error(e: blobstore::Error) -> Self {
+        Error {
+            status_code: StatusCode::BAD_GATEWAY,
+            message: format!("Error when communicating with blobstore: {}", e),
+        }
+    }
+
+    fn not_found() -> Self {
+        Error {
+            status_code: StatusCode::NOT_FOUND,
+            message: "Object not found".to_string(),
+        }
+    }
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+struct Blobby;
+
+const DEFAULT_CONTAINER_NAME: &str = "default";
+// TODO: Replace functionality for setting container name via header
+#[allow(dead_code)]
+const CONTAINER_HEADER_NAME: &str = "blobby-container";
+const CONTAINER_PARAM_NAME: &str = "container";
+
+/// A helper that will automatically create a container if it doesn't exist and returns an owned copy of the name for immediate use
+fn ensure_container(name: &String) -> Result<()> {
+    if !blobstore::container_exists(name).map_err(Error::from_blobstore_error)? {
+        blobstore::create_container(name).map_err(Error::from_blobstore_error)?;
+    }
+    Ok(())
+}
+
+fn send_response_error(response_out: ResponseOutparam, error: Error) {
+    let response = OutgoingResponse::new(Fields::new());
+    response
+        .set_status_code(error.status_code.as_u16())
+        .expect("Unable to set status code");
+    let response_body = response.body().unwrap();
+    response_body
+        .write()
+        .unwrap()
+        .blocking_write_and_flush(error.message.as_bytes())
+        .unwrap();
```

Comment:
apart from changing `unwrap`s to `expect` with clear panic messages, looking at this example could potentially signal to developers that this message is guaranteed to be fully sent (it's not) https://github.com/WebAssembly/wasi-io/blob/324be895965666805cb1c622ed4f071b9e3cbd65/wit/streams.wit#L134-L135

This operation should always be performed in a loop for byte lists of unknown size

Reply:
I'll just change this to use `write_all`